components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1264,6 +1264,26 @@ public static String getOAuth2IntrospectionEPUrl() {
                     OAuthServerConfiguration.getInstance()::getOauth2IntrospectionEPUrl);
         }
 
+        /**
+         * This method is used to get the resolved URL for the OAuth2 introspection endpoint.
+         *
+         * @param tenantDomain Tenant Domain.
+         * @return String of the resolved URL for the introspection endpoint.
+         * @throws URISyntaxException URI Syntax Exception.
+         */
+        public static String getOAuth2IntrospectionEPUrl(String tenantDomain) throws URISyntaxException {
+
+            String getOAuth2IntrospectionEPUrl = buildUrl(OAUTH2_INTROSPECT_EP_URL,
+                    OAuthServerConfiguration.getInstance()::getOauth2IntrospectionEPUrl);
+
+            if (!IdentityTenantUtil.isTenantQualifiedUrlsEnabled() && isNotSuperTenant(tenantDomain)) {
+                // Append tenant domain to path when the tenant-qualified url mode is disabled.
+                getOAuth2IntrospectionEPUrl =
```

Comment:
Why do we need the tenantDomain as a query param in the introspection endpoint in legacy mode?

Reply:
With this PR, the tenant domain added as a path param to the introspection endpoint in the legacy tenant URL mode.